app/main.py

```diff
@@ -1,2 +1,107 @@
 class Dictionary:
-    pass
+    def __init__(self):
+        self.ls = [None] * 8
+        self.capacity = 8
+        self.size = 0
+
+    def __setitem__(self, key, value):
+        if self.size == int(self.capacity * 2 / 3):
+            self.resize()
+        self.add_element(key, value)
+
+    def add_element(self, key, value):
+        key_hash = hash(key)
+        index = key_hash % self.capacity
+        if self.ls[index] is not None:
+            while True:
+                if index == self.capacity:
+                    index = 0
+                if self.ls[index] is None:
+                    self.ls[index] = (key, value, key_hash)
+                    self.size += 1
+                    break
+                if self.ls[index][0] == key:
+                    self.ls[index] = (key, value, key_hash)
+                    break
+                index += 1
+        else:
+            self.ls[index] = (key, value, key_hash)
+            self.size += 1
+
+    def resize(self):
+        self.capacity *= 2
+        copied_ls = self.ls.copy()
+        self.ls = [None] * self.capacity
+        self.size = 0
+        for cell in copied_ls:
+            if cell is not None:
+                self.add_element(cell[0], cell[1])
+
+    def __getitem__(self, key):
+        key_hash = hash(key)
+        index = key_hash % self.capacity
+        if self.ls[index] is None:
+            raise KeyError("No value with such key in this dict.")
```

Comment:
What if there are a few different keys with the same `key_hash % self.capacity`? `self.ls[index]` will not be `None` but the required key is missing. Rework.